tests/signature_aggregator_api.go

```diff
@@ -122,110 +118,20 @@ func SignatureAggregatorAPI(network *network.LocalNetwork, teleporter utils.Tele
-			m.name,
-			m.op,
-			m.value,
-		)
-	}
-
-	// make a second request, and ensure that the metrics reflect that the
-	// signatures for the second request are retrieved from the cache. note
-	// that even though 4 signatures were requested in the previous
-	// request, only 3 will be cached, because that's all that was required
-	// to reach a quorum, so that's all that were handled.
-	sendRequestToAPI()
-	metricsSample2 := sampleMetrics(signatureAggregatorConfig.MetricsPort)
-	Expect(
-		metricsSample2[metrics.Opts.AppRequestCount.Name],
-	).Should(Equal(metricsSample[metrics.Opts.AppRequestCount.Name]))
-	Expect(
-		metricsSample2[metrics.Opts.SignatureCacheHits.Name],
-	).Should(BeNumerically("==", 3))
-	Expect(
-		metricsSample2[metrics.Opts.SignatureCacheMisses.Name],
-	).Should(Equal(metricsSample[metrics.Opts.SignatureCacheMisses.Name]))
-}
-
-// returns a map of metric names to metric samples
-func sampleMetrics(port uint16) map[string]uint64 {
-	resp, err := http.Get(
-		fmt.Sprintf("http://localhost:%d/metrics", port),
+	// Try in the other direction
+	log.Info("Sending teleporter message from B -> A")
+	receipt, _, _ = testUtils.SendBasicTeleporterMessage(
+		ctx,
+		teleporter,
+		subnetBInfo,
+		subnetAInfo,
+		fundedKey,
+		fundedAddress,
 	)
-	Expect(err).Should(BeNil())
-
-	body, err := io.ReadAll(resp.Body)
-	Expect(err).Should(BeNil())
-	defer resp.Body.Close()
-
-	var samples = make(map[string]uint64)
-	scanner := bufio.NewScanner(strings.NewReader(string(body)))
-	for scanner.Scan() {
-		line := scanner.Text()
-		for _, metricName := range []string{
-			metrics.Opts.AggregateSignaturesLatencyMS.Name,
-			metrics.Opts.AggregateSignaturesRequestCount.Name,
-			metrics.Opts.AppRequestCount.Name,
-			metrics.Opts.FailuresToGetValidatorSet.Name,
-			metrics.Opts.FailuresToConnectToSufficientStake.Name,
-			metrics.Opts.FailuresSendingToNode.Name,
-			metrics.Opts.ValidatorTimeouts.Name,
-			metrics.Opts.InvalidSignatureResponses.Name,
-			metrics.Opts.SignatureCacheHits.Name,
-			metrics.Opts.SignatureCacheMisses.Name,
-			metrics.Opts.ConnectedStakeWeightPercentage.Name,
-		} {
-			if strings.HasPrefix(
-				line,
-				"U__signature_2d_aggregator_"+metricName,
-			) {
-				log.Debug("Found metric line", "line", line)
-				parts := strings.Fields(line)
-
-				metricName = strings.Replace(parts[0], "U__signature_2d_aggregator_", "", 1)
-
-				// Parse the metric count from the last field of the line
-				value, err := strconv.ParseUint(parts[len(parts)-1], 10, 64)
-				if err != nil {
-					log.Warn("failed to parse value from metric line")
-					continue
-				}
-				log.Debug("parsed metric", "name", metricName, "value", value)
-
-				samples[metricName] = value
-			} else {
-				log.Debug("Ignoring non-metric line", "line", line)
-			}
-		}
+	warpMessage = getWarpMessageFromLog(ctx, receipt, subnetBInfo)
+
+	reqBody = api.AggregateSignatureRequest{
+		Message: "0x" + hex.EncodeToString(warpMessage.Bytes()),
```

Comment:
We don't have a helper for prefixing with a `0x`!?

Reply:
There's some helpers here and there in various dependencies, but nothing immediately available that I'm able to find. We only do this a couple times and then only in tests, so I'm not too concerned.